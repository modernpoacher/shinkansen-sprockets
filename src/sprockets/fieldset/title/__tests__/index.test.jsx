import React from 'react'
import renderer from 'react-test-renderer'

import classnames from 'classnames'

import Super from '#sprockets/components/title'
import Title from '#sprockets/sprockets/fieldset/title'

jest.mock('classnames', () => jest.fn(() => 'MOCK CLASSNAME'))

describe('#sprockets/sprockets/fieldset/title', () => {
  describe('<Title />', () => {
    describe('With required props', () => {
      const component = (
        <Title />
      )

      it('renders', () => {
        return expect(renderer.create(component).toJSON())
          .toMatchSnapshot()
      })

      describe('`getClassName`', () => {
        it('is defined', () => {
          return expect(Title.prototype.getClassName)
            .toBeDefined()
        })
      })
    })

    describe('With additional props', () => {
      it('renders', () => {
        jest.spyOn(Title.prototype, 'hasTextContent').mockReturnValue(true)
        jest.spyOn(Title.prototype, 'getTextContent')
        jest.spyOn(Title.prototype, 'renderTextContent').mockReturnValue('MOCK RENDER CONTENT')

        const component = (
          <Title
            title='MOCK TITLE'
            onChange={jest.fn()}
          />
        )

        return expect(renderer.create(component).toJSON())
          .toMatchSnapshot()
      })
    })

    describe('`getClassName()`', () => {
      let returnValue

      beforeEach(() => {
        jest.spyOn(Super.prototype, 'getClassName').mockReturnValue('MOCK GETCLASSNAME')

        const component = (
          <Title />
        )

        const instance = (
          renderer.create(component)
            .getInstance()
        )

        returnValue = instance.getClassName()
      })

      it('invokes `classnames`', () => {
        return expect(classnames)
          .toBeCalledWith('MOCK GETCLASSNAME', 'fieldset')
      })

      it('returns the classname', () => {
        return expect(returnValue)
          .toBe('MOCK CLASSNAME')
      })
    })
  })
})
