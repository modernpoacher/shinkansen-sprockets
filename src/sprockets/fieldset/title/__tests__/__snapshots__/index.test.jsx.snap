// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@modernpoacher/sprockets/sprockets/fieldset/title <Title /> With additional props renders 1`] = `
<legend
  className="MOCK CLASSNAME"
>
  MOCK RENDER CONTENT
</legend>
`;

exports[`@modernpoacher/sprockets/sprockets/fieldset/title <Title /> With required props renders 1`] = `null`;
